conans/test/utils/tools.py

```diff
@@ -985,15 +986,30 @@ class GenConanfile(object):
 
     def __init__(self):
         self._imports = ["from conans import ConanFile"]
+        self._name = None
+        self._version = None
         self._settings = []
         self._options = {}
+        self._generators = []
         self._default_options = {}
         self._package_files = {}
         self._package_files_env = {}
         self._build_messages = []
         self._scm = {}
+        self._requires = []
         self._requirements = []
+        self._build_requirements = []
         self._revision_mode = None
+        self._package_info = {}
+        self._package_id_lines = []
+
+    def with_name(self, name):
```

Comment:
what's the advantage of using chained `with_xxx` calls instead of ctor args?
for me, it looks much more clean and natural (pythonic) to use just ctor, e.g.
```
GenConanfile(name="Pkg", version="0.1")
```
vs
```
GenConanfile().with_name("Pkg").with_version("0.1")
```

Reply:
I had that same doubt, but in this PR I've just followed the existing pattern. Probably it would be interesting to use a _builder_ pattern (`GenConanfile()...with_y()....build()`) or an instance based one (`GenConanfile(name=...,).with_xxxx()`) with some arguments in the ctor. Right now we are mixing them and there are things we cannot do.